samcli/commands/local/lib/swagger/reader.py

```diff
@@ -6,7 +6,7 @@
 import tempfile
 import logging
 
-from six.moves.urllib.parse import urlparse, parse_qs  # pylint: disable=relative-import
+from six.moves.urllib.parse import urlparse, parse_qs
```

Comment:
Do we need this anymore? Six is only used for py2 to 3 compatibility.

Reply:
Can we do this in another PR? I tried to remove all `six` usage and found there are a lot of them. This PR is already large in its size, I feel cutting the scope here makes review easier.

Reply:
Sure. We should probably do this soon though. Looks like six is not in our requirements file, which is concerning.

Reply:
Yeah, I found PyCharm also gave me this warning, looks like some of our dependencies depends on `six`, so my `pip freeze` still shows `six` installed.
I can take on it after getting this one merged.

Reply:
FYI:

```
six==1.15.0
  - astroid==2.4.2 [requires: six~=1.12]
    - pylint==2.3.1 [requires: astroid>=2.2.0,<3]
  - aws-lambda-builders==1.1.0 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: aws-lambda-builders==1.1.0]
  - aws-sam-translator==1.27.0 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
  - docker==4.2.2 [requires: six>=1.4.0]
    - aws-sam-cli==1.6.2 [requires: docker~=4.2.0]
  - jsonschema==3.2.0 [requires: six>=1.11.0]
    - aws-sam-translator==1.27.0 [requires: jsonschema~=3.0]
      - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
  - packaging==20.4 [requires: six]
    - pytest==5.2.1 [requires: packaging]
      - pytest-cov==2.7.1 [requires: pytest>=3.6]
      - pytest-forked==1.1.3 [requires: pytest>=3.1.0]
        - pytest-xdist==1.30.0 [requires: pytest-forked]
      - pytest-rerunfailures==7.0 [requires: pytest>=3.10]
      - pytest-timeout==1.3.3 [requires: pytest>=3.6.0]
      - pytest-xdist==1.30.0 [requires: pytest>=4.4.0]
  - pytest-xdist==1.30.0 [requires: six]
  - python-dateutil==2.8.0 [requires: six>=1.5]
    - arrow==0.16.0 [requires: python-dateutil>=2.7.0]
      - jinja2-time==0.2.0 [requires: arrow]
        - cookiecutter==1.6.0 [requires: jinja2-time>=0.1.0]
          - aws-sam-cli==1.6.2 [requires: cookiecutter~=1.6.0]
    - aws-sam-cli==1.6.2 [requires: python-dateutil~=2.6,<2.8.1]
    - botocore==1.17.63 [requires: python-dateutil>=2.1,<3.0.0]
      - boto3==1.14.63 [requires: botocore>=1.17.63,<1.18.0]
        - aws-sam-cli==1.6.2 [requires: boto3~=1.14.0,>=1.14.23]
        - aws-sam-translator==1.27.0 [requires: boto3~=1.5]
          - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
        - serverlessrepo==0.1.9 [requires: boto3~=1.9,>=1.9.56]
          - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
      - s3transfer==0.3.3 [requires: botocore>=1.12.36,<2.0a.0]
        - boto3==1.14.63 [requires: s3transfer>=0.3.0,<0.4.0]
          - aws-sam-cli==1.6.2 [requires: boto3~=1.14.0,>=1.14.23]
          - aws-sam-translator==1.27.0 [requires: boto3~=1.5]
            - aws-sam-cli==1.6.2 [requires: aws-sam-translator==1.27.0]
          - serverlessrepo==0.1.9 [requires: boto3~=1.9,>=1.9.56]
            - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
    - dateparser==0.7.6 [requires: python-dateutil]
      - aws-sam-cli==1.6.2 [requires: dateparser~=0.7]
  - serverlessrepo==0.1.9 [requires: six~=1.11]
    - aws-sam-cli==1.6.2 [requires: serverlessrepo==0.1.9]
  - websocket-client==0.57.0 [requires: six]
    - docker==4.2.2 [requires: websocket-client>=0.32.0]
      - aws-sam-cli==1.6.2 [requires: docker~=4.2.0]
```

Reply:
ya it is coming in as a dependency but if that gets dropped, the cli will explode.

It is not here: https://github.com/aws/aws-sam-cli/blob/develop/requirements/base.txt

Can we add six back in the meantime and then do the follow up with removing six?

Reply:
good idea, done